### PR TITLE
[SPIR-V] Support for multiple DebugCompilationUnit in DI

### PR DESCRIPTION
**Pull request by bwlodarcz**
The module in LLVM can have more then one CompilationUnit when e.g. modules are combined by llvm-linker.
 This property also needs to be handled in DI.

### CODE REVIEW

**Comment by bwlodarcz** [comment]
@michalpaszkowski @VyacheslavLevytskyy @Keenuts @MrSidims Please review. Thank you.

File: llvm/lib/Target/SPIRV/SPIRVEmitNonSemanticDI.cpp [hunk 1]
```diff
@@ -101,9 +101,10 @@ bool SPIRVEmitNonSemanticDI::emitGlobalDI(MachineFunction &MF) {
     for (const auto *Op : DbgCu->operands()) {
       if (const auto *CompileUnit = dyn_cast<DICompileUnit>(Op)) {
         DIFile *File = CompileUnit->getFile();
-        sys::path::append(FilePath, File->getDirectory(), File->getFilename());
-        SourceLanguage = CompileUnit->getSourceLanguage();
-        break;
+        FilePaths.emplace_back();
+        sys::path::append(FilePaths.back(), File->getDirectory(),
+                          File->getFilename());
+        SourceLanguages.push_back(CompileUnit->getSourceLanguage());
```
Comment (MrSidims) [inline llvm/lib/Target/SPIRV/SPIRVEmitNonSemanticDI.cpp:107]: Unrelated to this very question, but still. getSourceLanguage will return an integer for a source language known in LLVM's context. But it's not necessarily the same as defined in SPIR-V specification. So here should be a map to SPIR-V source languages defaulting to 3.2. Source Language: unknown.

Comment (bwlodarcz) [reply llvm/lib/Target/SPIRV/SPIRVEmitNonSemanticDI.cpp]: Done.

**Review by MrSidims — APPROVED** [review]
LGTM

**Review by michalpaszkowski — APPROVED** [review]
LGTM! Thanks for the pull request! If there are no more review comments, I will merge the change by tomorrow's SPIR-V Backend WG meeting time.

File: llvm/test/CodeGen/SPIRV/debug-info/debug-compilation-unit.ll [hunk 2]
```diff
@@ -6,21 +6,29 @@
...
-; CHECK-MIR-DAG: [[filename_str:%[0-9]+\:id\(s32\)]] = OpString 1094795567, 1094795585, 792805697, 1111638594, 1111638594, 1128481583, 1128481603, {{1697596227|1700545347}}, 1886216568, 1663985004, 0
-; CHECK-MIR-DAG: [[debug_source:%[0-9]+\:id\(s32\)]] = OpExtInst [[type_void]], 3, 35, [[filename_str]]
-; CHECK-MIR-DAG: [[debug_compilation_unit:%[0-9]+\:id\(s32\)]] = OpExtInst [[type_void]], 3, 1, [[source_language]], [[dwarf_version]], [[debug_source]], [[debug_info_version]]
+; CHECK-MIR-DAG: [[debug_info_version:%[0-9]+\:iid\(s32\)]] = OpConstantI [[type_i64]], 3
+; CHECK-MIR-DAG: [[source_language_sycl:%[0-9]+\:iid\(s32\)]] = OpConstantI [[type_i64]], 7
+; CHECK-MIR-DAG: [[source_language_cpp:%[0-9]+\:iid\(s32\)]] = OpConstantI [[type_i64]], 4
+; CHECK-MIR-DAG: [[filename_str_sycl:%[0-9]+\:id\(s32\)]] = OpString 1094795567, 1094795585, 792805697, 1111638594, 1111638594, 1128481583, 1128481603, {{1697596227|1700545347}}, 1886216568, 1663985004, 0
+; CHECK-MIR-DAG: [[filename_str_cpp:%[0-9]+\:id\(s32\)]] = OpString 1145324591, 1145324612, 793003076, 1162167621, 1162167621, 1179010607, 1179010630, 1697596998, 1886216568, 774989164, 7368803
```
Comment (tex3d) [inline llvm/test/CodeGen/SPIRV/debug-info/debug-compilation-unit.ll:13]: This broke on Windows (`/` vs. `\\`).  Fixed line should be:
```
; CHECK-MIR-DAG: [[filename_str_cpp:%[0-9]+\:id\(s32\)]] = OpString 1145324591, 1145324612, 793003076, 1162167621, 1162167621, 1179010607, 1179010630, {{1697596998|1700546118}}, 1886216568, 774989164, 7368803
```

Comment (tex3d) [reply llvm/test/CodeGen/SPIRV/debug-info/debug-compilation-unit.ll:13]: @bwlodarcz @michalpaszkowski FYI

Comment (michalpaszkowski) [reply llvm/test/CodeGen/SPIRV/debug-info/debug-compilation-unit.ll:13]: Thanks @tex3d for catching this! Bertrand prepared the fix here: https://github.com/llvm/llvm-project/pull/110249 Thanks @bwlodarcz!

**Pull request merged**